plugin.rb

```diff
@@ -137,151 +136,103 @@
-  end
-  add_to_class(:category, :activity_pub_default_object_type) do
-    DiscourseActivityPub::AP::Actor::Group.type
-  end
-  add_to_class(:category, :activity_pub_follower_count) { activity_pub_followers.count }
 
   on(:site_setting_changed) do |name, old_val, new_val|
     if %i[activity_pub_enabled login_required].include?(name)
-      Category
-        .joins(
-          "LEFT JOIN category_custom_fields ON categories.id = category_custom_fields.category_id",
-        )
-        .where(
-          "category_custom_fields.name = 'activity_pub_enabled' AND category_custom_fields.value IS NOT NULL",
-        )
-        .each(&:activity_pub_publish_state)
+      DiscourseActivityPubActor.active.each { |actor| actor.model.activity_pub_publish_state }
     end
   end
 
-  add_to_serializer(:basic_category, :activity_pub_enabled) { object.activity_pub_enabled }
-  add_to_serializer(
-    :site_category,
-    :activity_pub_ready,
-    include_condition: -> { object.activity_pub_enabled },
-  ) { object.activity_pub_ready? }
-  add_to_serializer(
-    :site_category,
-    :activity_pub_actor,
-    include_condition: -> do
-      object.activity_pub_enabled && object.respond_to?(:activity_pub_actor)
-    end,
-  ) do
-    DiscourseActivityPub::BasicActorSerializer.new(object.activity_pub_actor, root: false).as_json
-  end
-  add_to_serializer(
-    :site_category,
-    :activity_pub_actor_exists,
-    include_condition: -> { object.respond_to?(:activity_pub_actor) },
-  ) { object.activity_pub_actor.present? }
-  add_to_serializer(
-    :category,
-    :activity_pub_actor,
-    include_condition: -> do
-      object.activity_pub_enabled && object.respond_to?(:activity_pub_actor)
-    end,
-  ) do
-    DiscourseActivityPub::BasicActorSerializer.new(object.activity_pub_actor, root: false).as_json
-  end
-
-  if respond_to?(:register_preloaded_category_custom_fields)
-    register_preloaded_category_custom_fields("activity_pub_enabled")
-    register_preloaded_category_custom_fields("activity_pub_ready")
-  else
-    # TODO: Drop the if-statement and this if-branch in Discourse v3.2
-    Site.preloaded_category_custom_fields << "activity_pub_enabled"
-    Site.preloaded_category_custom_fields << "activity_pub_ready"
-  end
-
-  register_modifier(:site_all_categories_cache_query) do |query|
-    query.includes(:activity_pub_actor)
-  end
-
-  DiscourseActivityPubActor::CUSTOM_FIELDS.each do |field|
-    custom_field = "activity_pub_#{field}"
-    add_to_serializer(
-      :basic_category,
-      custom_field.to_sym,
-      include_condition: -> { object.activity_pub_enabled },
-    ) { object.send(custom_field) }
-
-    if respond_to?(:register_preloaded_category_custom_fields)
-      register_preloaded_category_custom_fields(custom_field)
-    else
-      # TODO: Drop the if-statement and this if-branch in Discourse v3.2
-      Site.preloaded_category_custom_fields << custom_field
+  add_to_serializer(:site, :activity_pub_actors) do
```

Comment:
It might be nice long-term to move this outside of the site serializer. This could grow quite quickly on a site with many categories/tags that have AP enabled.